internal/integration/ogent/ogent_test.go

```diff
@@ -221,3 +232,70 @@ func (t *testSuite) reqErr(c int, err interface{}) {
 	t.Require().Equal(c, ac, err)
 	t.Require().Equal(http.StatusText(c), at, err)
 }
+
+func (t *testSuite) TestParseFilterToSQL() {
+	f, e := preparePetFilterToSQL()
+	t.Equal(len(f), len(e), "filter and expected list must have the same length")
+	i := 0
+	n := len(f)
+	for i < n {
+		q := t.client.Pet.Query()
+		exp := &ogent.Expression{}
+		err := ogent.SqlParser.ParseString(f[i], exp)
+		t.Require().NoError(err, "fail to parse the filter: "+f[i])
+		q.Where(func(s *sql.Selector) {
+			ogent.ParseExpression(exp, s)
+			query, _ := s.Query()
+			t.Equal(e[i], query, "actual query created is not the expected one")
+		})
+		_, err = q.All(context.Background())
+		t.Require().NoError(err, "fail to execute the query")
+		i++
+	}
+}
+
+func preparePetFilterToSQL() ([]string, []string) {
+	var f []string //filter
+	var e []string //expected
+	//EQ
+	f = append(f, "name eq 'Ariel'")
+	e = append(e, "SELECT `pets`.`id`, `pets`.`name`, `pets`.`weight`, `pets`.`birthday` FROM `pets` WHERE `pets`.`name` = ?")
+	//NE
+	f = append(f, "name ne 'Milk'")
+	e = append(e, "SELECT `pets`.`id`, `pets`.`name`, `pets`.`weight`, `pets`.`birthday` FROM `pets` WHERE `pets`.`name` <> ?")
+	//GT
+	f = append(f, "weight gt 2")
```

Comment:
Can you elaborate on this syntax? I created something in the past named [rql](https://github.com/a8m/rql) which uses a mongo-like syntax. @masseelch, may want to give it a look. I tried to use something common at that time. 

IMO, the syntax for the GraphQL filters I mentioned above can be used here as well. It is generic enough and can be defined and used statically in the client code. It is also much easier to work with objects when constructing the filters on the client-side, and both client and backend code can be shared here.

Reply:
I updated the test cases